<div style=" background-color: RGB(0,114,200);" >
<h1 style="margin: auto; padding: 20px 0; color:#fff; text-align: center">PROJET 4 DATA ANALYST</h1>
<h2 style="margin: auto; padding: 20px 0; color:#fff; text-align: center">Réalisez une étude de santé publique avec R ou Python
</h2>
</div>

# OBJECTIF DE CE NOTEBOOK

Bienvenue dans l'outil plébiscité par les analystes de données Jupyter.

Il s'agit d'un outil permettant de mixer et d'alterner codes, textes et graphique.

Cet outil est formidable pour plusieurs raisons:

+ il permet de tester des lignes de codes au fur et à mesure de votre rédaction, de constater immédiatement le résultat d'un instruction, de la corriger si nécessaire.
+ De rédiger du texte pour expliquer l'approche suivie ou les résultats d'une analyse et de le mettre en forme grâce à du code html ou plus simple avec **Markdown**
+ d'agrémenter de graphiques

Pour vous aider dans vos premiers pas à l'usage de Jupyter et de Python, nous avons rédigé ce notebook en vous indiquant les instructions à suivre.

Il vous suffit pour cela de saisir le code Python répondant à l'instruction donnée.

Vous verrez de temps à autre le code Python répondant à une instruction donnée mais cela est fait pour vous aider à comprendre la nature du travail qui vous est demandée.

Et garder à l'esprit, qu'il n'y a pas de solution unique pour résoudre un problème et qu'il y a autant de résolutions de problèmes que de développeurs ;)...



<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Etape 1 - Importation des librairies et chargement des fichiers</h2>
</div>

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">1.1 - Importation des librairies</h3>
</div>

In [3]:
# Importation de la librairie Pandas
import pandas as pd

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">1.2 - Chargement des fichiers Excel</h3>
</div>

In [4]:
#Importation du fichier population.csv
population = pd.read_csv("data/population.csv")

#Importation du fichier dispo_alimentaire.csv
dispo = pd.read_csv("data/dispo_alimentaire.csv")

#Importation du fichier aide_alimentaire.csv
aide = pd.read_csv("data/aide_alimentaire.csv")

#Importation du fichier sous_nutrition.csv
sous_nutrition = pd.read_csv("data/sous_nutrition.csv")

<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Etape 2 - Analyse exploratoire des fichiers</h2>
</div>

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.1 - Analyse exploratoire du fichier population</h3>
</div>

In [5]:
#Afficher les dimensions du dataset
print("Le tableau comporte {} observation(s) ou article(s)".format(population.shape[0]))
print("Le tableau comporte {} colonne(s)".format(population.shape[1]))

Le tableau comporte 1416 observation(s) ou article(s)
Le tableau comporte 3 colonne(s)


In [6]:
#Consulter le nombre de colonnes
#La nature des données dans chacune des colonnes
#Le nombre de valeurs présentes dans chacune des colonnes

# Informations sur le fichier population.csv
print("\nPopulation :")
print(population.info())
print(population.describe())

# Informations sur dispo_alimentaire.csv
print("\nDisponibilité alimentaire :")
print(dispo.info())
print(dispo.describe())

# Informations sur aide_alimentaire.csv
print("\nAide alimentaire :")
print(aide.info())
print(aide.describe())

# Informations sur sous_nutrition.csv
print("\nSous-nutrition :")
print(sous_nutrition.info())
print(sous_nutrition.describe())


Population :
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1416 entries, 0 to 1415
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Zone    1416 non-null   object 
 1   Année   1416 non-null   int64  
 2   Valeur  1416 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 33.3+ KB
None
             Année        Valeur
count  1416.000000  1.416000e+03
mean   2015.500000  3.144793e+04
std       1.708428  1.300812e+05
min    2013.000000  7.930000e-01
25%    2014.000000  3.783410e+02
50%    2015.500000  5.126480e+03
75%    2017.000000  1.930666e+04
max    2018.000000  1.427648e+06

Disponibilité alimentaire :
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15605 entries, 0 to 15604
Data columns (total 18 columns):
 #   Column                                                         Non-Null Count  Dtype  
---  ------                                                         --------------  -----  
 0   Zon

In [7]:
#Affichage les 5 premières lignes de la table

# Affichage du fichier population.csv
print("Population :")
display(population.head())

# Affichage du fichier dispo_alimentaire.csv
print("Disponibilité alimentaire :")
display(dispo.head())

# Affichage du fichier aide_alimentaire.csv
print("Aide alimentaire :")
display(aide.head())

# Affichage du fichier sous_nutrition.csv
print("Sous-nutrition :")
display(sous_nutrition.head())

Population :


Zone  Année     Valeur
0  Afghanistan   2013  32269.589
1  Afghanistan   2014  33370.794
2  Afghanistan   2015  34413.603
3  Afghanistan   2016  35383.032
4  Afghanistan   2017  36296.113

Disponibilité alimentaire :


Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    NaN   
1  Afghanistan        Agrumes, Autres  vegetale                    NaN   
2  Afghanistan  Aliments pour enfants  vegetale                    NaN   
3  Afghanistan                 Ananas  vegetale                    NaN   
4  Afghanistan                Bananes  vegetale                    NaN   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  NaN                                             5.0   
1                  NaN                                             1.0   
2                  NaN                                             1.0   
3                  NaN                                             0.0   
4                  NaN                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.29        
2                                               0.06        
3                                               0.00        
4                                               2.70        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               0.01               
2                                               0.01               
3                                                NaN               
4                                               0.02               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.02          
2                                               0.03          
3                                                NaN          
4                                               0.05          

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                      53.0                      NaN                      NaN   
1                      41.0                      2.0                     40.0   
2                       2.0                      NaN                      2.0   
3                       0.0                      NaN                      0.0   
4                      82.0                      NaN                     82.0   

   Nourriture  Pertes  Production  Semences  Traitement  Variation de stock  
0        53.0     NaN        53.0       NaN         NaN                 NaN  
1        39.0     2.0         3.0       NaN         NaN                 NaN  
2         2.0     NaN         NaN       NaN         NaN                 NaN  
3         0.0     NaN         NaN       NaN         NaN                 NaN  
4        82.0     NaN         NaN       NaN         NaN                 NaN

Aide alimentaire :


Pays bénéficiaire  Année              Produit  Valeur
0       Afghanistan   2013  Autres non-céréales     682
1       Afghanistan   2014  Autres non-céréales     335
2       Afghanistan   2013         Blé et Farin   39224
3       Afghanistan   2014         Blé et Farin   15160
4       Afghanistan   2013             Céréales   40504

Sous-nutrition :


Zone      Année Valeur
0  Afghanistan  2012-2014    8.6
1  Afghanistan  2013-2015    8.8
2  Afghanistan  2014-2016    8.9
3  Afghanistan  2015-2017    9.7
4  Afghanistan  2016-2018   10.5

In [8]:
#Nous allons harmoniser les unités. Pour cela, nous avons décidé de multiplier la population par 1000

# Multiplication de la colonne valeur par 1000
population["Valeur"] = population["Valeur"] * 1000
display(population.head())

Zone  Année      Valeur
0  Afghanistan   2013  32269589.0
1  Afghanistan   2014  33370794.0
2  Afghanistan   2015  34413603.0
3  Afghanistan   2016  35383032.0
4  Afghanistan   2017  36296113.0

In [9]:
#Changement du nom de la colonne Valeur par Population
population = population.rename(columns={"Valeur": "Population"})

In [10]:
#Affichage les 5 premières lignes de la table pour voir les modifications
display(population.head())

Zone  Année  Population
0  Afghanistan   2013  32269589.0
1  Afghanistan   2014  33370794.0
2  Afghanistan   2015  34413603.0
3  Afghanistan   2016  35383032.0
4  Afghanistan   2017  36296113.0

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.2 - Analyse exploratoire du fichier disponibilité alimentaire</h3>
</div>

In [11]:
#Afficher les dimensions du dataset
print("Dimensions du dataset :", dispo.shape)

Dimensions du dataset : (15605, 18)


In [12]:
#Consulter le nombre de colonnes
print("\nInfos sur les colonnes :")
print(dispo.info())


Infos sur les colonnes :
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15605 entries, 0 to 15604
Data columns (total 18 columns):
 #   Column                                                         Non-Null Count  Dtype  
---  ------                                                         --------------  -----  
 0   Zone                                                           15605 non-null  object 
 1   Produit                                                        15605 non-null  object 
 2   Origine                                                        15605 non-null  object 
 3   Aliments pour animaux                                          2720 non-null   float64
 4   Autres Utilisations                                            5496 non-null   float64
 5   Disponibilité alimentaire (Kcal/personne/jour)                 14241 non-null  float64
 6   Disponibilité alimentaire en quantité (kg/personne/an)         14015 non-null  float64
 7   Disponibilité de matière grasse 

In [13]:
#Affichage les 5 premières lignes de la table
print("\nAperçu du dataset :")
display(dispo.head())


Aperçu du dataset :


Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    NaN   
1  Afghanistan        Agrumes, Autres  vegetale                    NaN   
2  Afghanistan  Aliments pour enfants  vegetale                    NaN   
3  Afghanistan                 Ananas  vegetale                    NaN   
4  Afghanistan                Bananes  vegetale                    NaN   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  NaN                                             5.0   
1                  NaN                                             1.0   
2                  NaN                                             1.0   
3                  NaN                                             0.0   
4                  NaN                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.29        
2                                               0.06        
3                                               0.00        
4                                               2.70        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               0.01               
2                                               0.01               
3                                                NaN               
4                                               0.02               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.02          
2                                               0.03          
3                                                NaN          
4                                               0.05          

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                      53.0                      NaN                      NaN   
1                      41.0                      2.0                     40.0   
2                       2.0                      NaN                      2.0   
3                       0.0                      NaN                      0.0   
4                      82.0                      NaN                     82.0   

   Nourriture  Pertes  Production  Semences  Traitement  Variation de stock  
0        53.0     NaN        53.0       NaN         NaN                 NaN  
1        39.0     2.0         3.0       NaN         NaN                 NaN  
2         2.0     NaN         NaN       NaN         NaN                 NaN  
3         0.0     NaN         NaN       NaN         NaN                 NaN  
4        82.0     NaN         NaN       NaN         NaN                 NaN

In [14]:
#Remplacement des NaN dans le dataset par des 0
dispo = dispo.fillna(0)
print("\nAprès remplacement des NaN par 0 :")
display(dispo.head())


Après remplacement des NaN par 0 :


Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    0.0   
1  Afghanistan        Agrumes, Autres  vegetale                    0.0   
2  Afghanistan  Aliments pour enfants  vegetale                    0.0   
3  Afghanistan                 Ananas  vegetale                    0.0   
4  Afghanistan                Bananes  vegetale                    0.0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  0.0                                             5.0   
1                  0.0                                             1.0   
2                  0.0                                             1.0   
3                  0.0                                             0.0   
4                  0.0                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.29        
2                                               0.06        
3                                               0.00        
4                                               2.70        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               0.01               
2                                               0.01               
3                                               0.00               
4                                               0.02               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.02          
2                                               0.03          
3                                               0.00          
4                                               0.05          

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                      53.0                      0.0                      0.0   
1                      41.0                      2.0                     40.0   
2                       2.0                      0.0                      2.0   
3                       0.0                      0.0                      0.0   
4                      82.0                      0.0                     82.0   

   Nourriture  Pertes  Production  Semences  Traitement  Variation de stock  
0        53.0     0.0        53.0       0.0         0.0                 0.0  
1        39.0     2.0         3.0       0.0         0.0                 0.0  
2         2.0     0.0         0.0       0.0         0.0                 0.0  
3         0.0     0.0         0.0       0.0         0.0                 0.0  
4        82.0     0.0         0.0       0.0         0.0                 0.0

In [15]:
#Multiplication de toutes les lignes contenant des milliers de tonnes en Kg
colonnes_tonnes = [
    "Disponibilité intérieure",
    "Aliments pour animaux",
    "Semences",
    "Pertes",
    "Transformations",
    "Exportations",
    "Nourriture",
    "Autres utilisations"
]

for col in colonnes_tonnes:
    if col in dispo.columns:
        dispo[col] = dispo[col] * 1000

In [16]:
#Affichage les 5 premières lignes de la table
print("\nDataset après conversion en kg :")
display(dispo.head())


Dataset après conversion en kg :


Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    0.0   
1  Afghanistan        Agrumes, Autres  vegetale                    0.0   
2  Afghanistan  Aliments pour enfants  vegetale                    0.0   
3  Afghanistan                 Ananas  vegetale                    0.0   
4  Afghanistan                Bananes  vegetale                    0.0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  0.0                                             5.0   
1                  0.0                                             1.0   
2                  0.0                                             1.0   
3                  0.0                                             0.0   
4                  0.0                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.29        
2                                               0.06        
3                                               0.00        
4                                               2.70        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               0.01               
2                                               0.01               
3                                               0.00               
4                                               0.02               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.02          
2                                               0.03          
3                                               0.00          
4                                               0.05          

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                   53000.0                      0.0                      0.0   
1                   41000.0                      2.0                     40.0   
2                    2000.0                      0.0                      2.0   
3                       0.0                      0.0                      0.0   
4                   82000.0                      0.0                     82.0   

   Nourriture  Pertes  Production  Semences  Traitement  Variation de stock  
0     53000.0     0.0        53.0       0.0         0.0                 0.0  
1     39000.0  2000.0         3.0       0.0         0.0                 0.0  
2      2000.0     0.0         0.0       0.0         0.0                 0.0  
3         0.0     0.0         0.0       0.0         0.0                 0.0  
4     82000.0     0.0         0.0       0.0         0.0                 0.0

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.3 - Analyse exploratoire du fichier aide alimentaire</h3>
</div>

In [17]:
#Afficher les dimensions du dataset
print("Dimensions du dataset :", aide.shape)

Dimensions du dataset : (1475, 4)


In [18]:
#Consulter le nombre de colonnes
print("\nInfos sur les colonnes :")
print(aide.info())


Infos sur les colonnes :
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1475 entries, 0 to 1474
Data columns (total 4 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Pays bénéficiaire  1475 non-null   object
 1   Année              1475 non-null   int64 
 2   Produit            1475 non-null   object
 3   Valeur             1475 non-null   int64 
dtypes: int64(2), object(2)
memory usage: 46.2+ KB
None


In [19]:
#Affichage les 5 premières lignes de la table
print("\nInfos sur les colonnes :")
print(aide.head())


Infos sur les colonnes :
  Pays bénéficiaire  Année              Produit  Valeur
0       Afghanistan   2013  Autres non-céréales     682
1       Afghanistan   2014  Autres non-céréales     335
2       Afghanistan   2013         Blé et Farin   39224
3       Afghanistan   2014         Blé et Farin   15160
4       Afghanistan   2013             Céréales   40504


In [20]:
#Changement du nom de la colonne Pays bénéficiaire par Zone
aide = aide.rename(columns={"Pays bénéficiaire": "Zone"})
# Vérification
print(aide.head())

          Zone  Année              Produit  Valeur
0  Afghanistan   2013  Autres non-céréales     682
1  Afghanistan   2014  Autres non-céréales     335
2  Afghanistan   2013         Blé et Farin   39224
3  Afghanistan   2014         Blé et Farin   15160
4  Afghanistan   2013             Céréales   40504


In [21]:
#Multiplication de la colonne Aide_alimentaire qui contient des tonnes par 1000 pour avoir des kg
aide["Valeur"] = aide["Valeur"] * 1000

In [22]:
#Affichage les 5 premières lignes de la table
display(aide.head())

Zone  Année              Produit    Valeur
0  Afghanistan   2013  Autres non-céréales    682000
1  Afghanistan   2014  Autres non-céréales    335000
2  Afghanistan   2013         Blé et Farin  39224000
3  Afghanistan   2014         Blé et Farin  15160000
4  Afghanistan   2013             Céréales  40504000

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.3 - Analyse exploratoire du fichier sous nutrition</h3>
</div>

In [23]:
#Afficher les dimensions du dataset
print("Dimensions du dataset :", sous_nutrition.shape)

Dimensions du dataset : (1218, 3)


In [24]:
#Consulter le nombre de colonnes
print("\nInfos sur les colonnes :")
print(sous_nutrition.info())


Infos sur les colonnes :
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1218 entries, 0 to 1217
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Zone    1218 non-null   object
 1   Année   1218 non-null   object
 2   Valeur  624 non-null    object
dtypes: object(3)
memory usage: 28.7+ KB
None


In [25]:
#Afficher les 5 premières lignes de la table
print("\nInfos sur les colonnes :")
print(sous_nutrition.info())


Infos sur les colonnes :
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1218 entries, 0 to 1217
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Zone    1218 non-null   object
 1   Année   1218 non-null   object
 2   Valeur  624 non-null    object
dtypes: object(3)
memory usage: 28.7+ KB
None


In [26]:
#Conversion de la colonne sous nutrition en numérique

# Remplacer la chaîne "<0.1>" par "0.1"
sous_nutrition['Valeur'] = sous_nutrition['Valeur'].replace("<0.1", "0.1")

# Conversion en numérique
sous_nutrition['Valeur'] = pd.to_numeric(sous_nutrition['Valeur'])

# Vérification
print(sous_nutrition['Valeur'].unique()[:20])


[ 8.6  8.8  8.9  9.7 10.5 11.1  2.2  2.5  2.8  3.   3.1  3.3  0.1  1.3
  1.2  nan  7.6  6.2  5.3  5.6]


In [27]:
#Conversion de la colonne (avec l'argument errors=coerce qui permet de convertir automatiquement les lignes qui ne sont pas des nombres en NaN)
#Puis remplacement des NaN en 0

# Conversion de la colonne Valeur en numérique
sous_nutrition['Valeur'] = pd.to_numeric(sous_nutrition['Valeur'], errors='coerce')

# Remplacement des NaN par 0
sous_nutrition['Valeur'] = sous_nutrition['Valeur'].fillna(0)

# Vérification
display(sous_nutrition.head())


Zone      Année  Valeur
0  Afghanistan  2012-2014     8.6
1  Afghanistan  2013-2015     8.8
2  Afghanistan  2014-2016     8.9
3  Afghanistan  2015-2017     9.7
4  Afghanistan  2016-2018    10.5

In [28]:
#changement du nom de la colonne Valeur par sous_nutrition
sous_nutrition =sous_nutrition.rename(columns={"Valeur": "sous_nutrition"})

In [29]:
#Multiplication de la colonne sous_nutrition par 1000000
sous_nutrition["sous_nutrition"] = sous_nutrition["sous_nutrition"] * 1000

In [30]:
#Afficher les 5 premières lignes de la table
display(sous_nutrition.head())

Zone      Année  sous_nutrition
0  Afghanistan  2012-2014          8600.0
1  Afghanistan  2013-2015          8800.0
2  Afghanistan  2014-2016          8900.0
3  Afghanistan  2015-2017          9700.0
4  Afghanistan  2016-2018         10500.0

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.1 - Proportion de personnes en sous nutrition</h3>
</div>

In [31]:
#Il faut tout d'abord faire une jointure entre la table population et la table sous nutrition, en ciblant l'année 2017

# Filtrer population 2017
popu_2017 = population[population["Année"] == 2017]

# Filtrer sous_nutrition sur la période 2016-2018 (qui couvre 2017)
sn_2017 = sous_nutrition[sous_nutrition["Année"] == "2016-2018"]

# Merge sur la colonne Zone
popu_nutri = pd.merge(sn_2017, popu_2017, on="Zone", how="inner")

In [32]:
#Affichage du dataset
display(popu_nutri.head(20))

Zone    Année_x  sous_nutrition  Année_y   Population
0          Afghanistan  2016-2018         10500.0     2017   36296113.0
1       Afrique du Sud  2016-2018          3100.0     2017   57009756.0
2              Albanie  2016-2018           100.0     2017    2884169.0
3              Algérie  2016-2018          1300.0     2017   41389189.0
4            Allemagne  2016-2018             0.0     2017   82658409.0
5              Andorre  2016-2018             0.0     2017      77001.0
6               Angola  2016-2018          5800.0     2017   29816766.0
7   Antigua-et-Barbuda  2016-2018             0.0     2017      95426.0
8      Arabie saoudite  2016-2018          1600.0     2017   33101179.0
9            Argentine  2016-2018          1500.0     2017   43937140.0
10             Arménie  2016-2018           100.0     2017    2944791.0
11           Australie  2016-2018             0.0     2017   24584620.0
12            Autriche  2016-2018             0.0     2017    8819901.0
13         Azerbaïdjan  2016-2018             0.0     2017    9845320.0
14             Bahamas  2016-2018             0.0     2017     381755.0
15             Bahreïn  2016-2018             0.0     2017    1494076.0
16          Bangladesh  2016-2018         21500.0     2017  159685424.0
17             Barbade  2016-2018           100.0     2017     286232.0
18             Bélarus  2016-2018             0.0     2017    9450231.0
19            Belgique  2016-2018             0.0     2017   11419748.0

In [33]:
#Calcul et affichage du nombre de personnes en état de sous nutrition

# Réorganisation des colonnes et renommage
popu_nutri = popu_nutri[["Zone", "Année_y", "Population", "sous_nutrition"]].rename(columns={"Année_y": "Année"})

# Calcul de la proportion par pays
popu_nutri["Proportion_sous_nutrition"] = (popu_nutri["sous_nutrition"] / popu_nutri["Population"]) * 100

# Calcul de la proportion mondiale
prop_mondiale = popu_nutri["sous_nutrition"].sum() / popu_nutri["Population"].sum() * 100

# Affichage du résultat
print("Proportion mondiale de sous-nutrition en 2017 : {:.2f}%".format(prop_mondiale))
display(popu_nutri.head(20))

Proportion mondiale de sous-nutrition en 2017 : 0.01%


Zone  Année   Population  sous_nutrition  \
0          Afghanistan   2017   36296113.0         10500.0   
1       Afrique du Sud   2017   57009756.0          3100.0   
2              Albanie   2017    2884169.0           100.0   
3              Algérie   2017   41389189.0          1300.0   
4            Allemagne   2017   82658409.0             0.0   
5              Andorre   2017      77001.0             0.0   
6               Angola   2017   29816766.0          5800.0   
7   Antigua-et-Barbuda   2017      95426.0             0.0   
8      Arabie saoudite   2017   33101179.0          1600.0   
9            Argentine   2017   43937140.0          1500.0   
10             Arménie   2017    2944791.0           100.0   
11           Australie   2017   24584620.0             0.0   
12            Autriche   2017    8819901.0             0.0   
13         Azerbaïdjan   2017    9845320.0             0.0   
14             Bahamas   2017     381755.0             0.0   
15             Bahreïn   2017    1494076.0             0.0   
16          Bangladesh   2017  159685424.0         21500.0   
17             Barbade   2017     286232.0           100.0   
18             Bélarus   2017    9450231.0             0.0   
19            Belgique   2017   11419748.0             0.0   

    Proportion_sous_nutrition  
0                    0.028929  
1                    0.005438  
2                    0.003467  
3                    0.003141  
4                    0.000000  
5                    0.000000  
6                    0.019452  
7                    0.000000  
8                    0.004834  
9                    0.003414  
10                   0.003396  
11                   0.000000  
12                   0.000000  
13                   0.000000  
14                   0.000000  
15                   0.000000  
16                   0.013464  
17                   0.034937  
18                   0.000000  
19                   0.000000

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.2 - Nombre théorique de personne qui pourrait être nourries</h3>
</div>

In [34]:
#Combien mange en moyenne un être humain ? Source => FAO ≈ 2500 kcal/jour
besoin_kcal = 2500

In [35]:
#On commence par faire une jointure entre le data frame population et dispo_alimentaire afin d'ajouter dans ce dernier la population

# Filtrer la population pour 2017
popu_2017 = population[population["Année"] == 2017]

# Faire la jointure
dispo_pop = pd.merge(dispo, popu_2017, on="Zone", how="inner")


In [36]:
#Affichage du nouveau dataframe
display(dispo_pop.head())

Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    0.0   
1  Afghanistan        Agrumes, Autres  vegetale                    0.0   
2  Afghanistan  Aliments pour enfants  vegetale                    0.0   
3  Afghanistan                 Ananas  vegetale                    0.0   
4  Afghanistan                Bananes  vegetale                    0.0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  0.0                                             5.0   
1                  0.0                                             1.0   
2                  0.0                                             1.0   
3                  0.0                                             0.0   
4                  0.0                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1.72        
1                                               1.29        
2                                               0.06        
3                                               0.00        
4                                               2.70        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                               0.20               
1                                               0.01               
2                                               0.01               
3                                               0.00               
4                                               0.02               

   Disponibilité de protéines en quantité (g/personne/jour)  \
0                                               0.77          
1                                               0.02          
2                                               0.03          
3                                               0.00          
4                                               0.05          

   Disponibilité intérieure  Exportations - Quantité  Importations - Quantité  \
0                   53000.0                      0.0                      0.0   
1                   41000.0                      2.0                     40.0   
2                    2000.0                      0.0                      2.0   
3                       0.0                      0.0                      0.0   
4                   82000.0                      0.0                     82.0   

   Nourriture  Pertes  Production  Semences  Traitement  Variation de stock  \
0     53000.0     0.0        53.0       0.0         0.0                 0.0   
1     39000.0  2000.0         3.0       0.0         0.0                 0.0   
2      2000.0     0.0         0.0       0.0         0.0                 0.0   
3         0.0     0.0         0.0       0.0         0.0                 0.0   
4     82000.0     0.0         0.0       0.0         0.0                 0.0   

   Année  Population  
0   2017  36296113.0  
1   2017  36296113.0  
2   2017  36296113.0  
3   2017  36296113.0  
4   2017  36296113.0

In [37]:
#Création de la colonne dispo_kcal avec calcul des kcal disponibles mondialement
dispo_pop["dispo_kcal"] = dispo_pop["Disponibilité alimentaire (Kcal/personne/jour)"] * dispo_pop["Population"] * 365

# Somme mondiale
total_kcal = dispo_pop["dispo_kcal"].sum()
print("Total des kcal disponibles mondialement en 2017 :", total_kcal)

Total des kcal disponibles mondialement en 2017 : 7635429388975815.0


In [38]:
#Calcul du nombre d'humains pouvant être nourris
nb_humains_nourris = total_kcal / (besoin_kcal * 365)  # sur une année
print("Nombre théorique d'humains pouvant être nourris en 2017 :", int(nb_humains_nourris))

Nombre théorique d'humains pouvant être nourris en 2017 : 8367593850


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.3 - Nombre théorique de personne qui pourrait être nourrie avec les produits végétaux</h3>
</div>

In [39]:
# Transfert des données avec les végétaux dans un nouveau dataframe
pd.set_option('display.float_format', '{:,.0f}'.format)

dispo_vegetaux = dispo_pop[dispo_pop["Origine"] == "vegetale"]
display(dispo_vegetaux.head())

Zone                Produit   Origine  Aliments pour animaux  \
1  Afghanistan        Agrumes, Autres  vegetale                      0   
2  Afghanistan  Aliments pour enfants  vegetale                      0   
3  Afghanistan                 Ananas  vegetale                      0   
4  Afghanistan                Bananes  vegetale                      0   
6  Afghanistan                  Bière  vegetale                      0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
1                    0                                               1   
2                    0                                               1   
3                    0                                               0   
4                    0                                               4   
6                    0                                               0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
1                                                  1        
2                                                  0        
3                                                  0        
4                                                  3        
6                                                  0        

   Disponibilité de matière grasse en quantité (g/personne/jour)  \
1                                                  0               
2                                                  0               
3                                                  0               
4                                                  0               
6                                                  0               

   Disponibilité de protéines en quantité (g/personne/jour)  \
1                                                  0          
2                                                  0          
3                                                  0          
4                                                  0          
6                                                  0          

   Disponibilité intérieure  ...  Importations - Quantité  Nourriture  Pertes  \
1                    41,000  ...                       40      39,000   2,000   
2                     2,000  ...                        2       2,000       0   
3                         0  ...                        0           0       0   
4                    82,000  ...                       82      82,000       0   
6                     3,000  ...                        3       3,000       0   

   Production  Semences  Traitement  Variation de stock  Année  Population  \
1           3         0           0                   0   2017  36,296,113   
2           0         0           0                   0   2017  36,296,113   
3           0         0           0                   0   2017  36,296,113   
4           0         0           0                   0   2017  36,296,113   
6           0         0           0                   0   2017  36,296,113   

      dispo_kcal  
1 13,248,081,245  
2 13,248,081,245  
3              0  
4 52,992,324,980  
6              0  

[5 rows x 21 columns]

In [40]:
# Calcul du nombre de kcal disponible pour les végétaux
dispo_vegetaux = dispo_pop[dispo_pop["Origine"] == "vegetale"].copy()

dispo_vegetaux["dispo_kcal_veg"] = (
    dispo_vegetaux["Disponibilité alimentaire (Kcal/personne/jour)"]
    * dispo_vegetaux["Population"]
    * 365
)

# Somme mondiale
total_kcal_veg = dispo_vegetaux["dispo_kcal_veg"].sum()
print("Total des kcal disponibles provenant des produits végétaux en 2017 :", total_kcal_veg)

Total des kcal disponibles provenant des produits végétaux en 2017 : 6300178937197865.0


In [41]:
# Calcul du nombre d'humains pouvant être nourris avec les végétaux
nb_humains_veg = total_kcal_veg / (besoin_kcal * 365)
print("Nombre théorique d'humains pouvant être nourris avec les produits végétaux en 2017 :", int(nb_humains_veg))


Nombre théorique d'humains pouvant être nourris avec les produits végétaux en 2017 : 6904305684


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.4 - Utilisation de la disponibilité intérieure</h3>
</div>

In [42]:
# Calcul de la disponibilité totale mondiale (somme de la colonne 'Disponibilité intérieure - Quantité')
dispo_totale = dispo_pop["Disponibilité intérieure"].sum()
print("Disponibilité intérieure totale mondiale :", dispo_totale)

Disponibilité intérieure totale mondiale : 9733927000.0


In [43]:
# Création d'une boucle pour afficher les différentes utilisations de la disponibilité intérieure
colonnes_utilisation = [
    "Aliments pour animaux",
    "Pertes",
    "Nourriture",
    "Semences",
    "Traitement",
    "Autres Utilisations"
]

for col in colonnes_utilisation:
    utilisation = dispo_pop[col].sum()
    pourcentage = (utilisation / dispo_totale) * 100
    print(f"{col} : {utilisation:.2f} tonnes, soit {pourcentage:.2f}% de la disponibilité totale")

Aliments pour animaux : 1288002000.00 tonnes, soit 13.23% de la disponibilité totale
Pertes : 452283000.00 tonnes, soit 4.65% de la disponibilité totale
Nourriture : 4805525000.00 tonnes, soit 49.37% de la disponibilité totale
Semences : 153317000.00 tonnes, soit 1.58% de la disponibilité totale
Traitement : 2185641.00 tonnes, soit 0.02% de la disponibilité totale
Autres Utilisations : 858771.00 tonnes, soit 0.01% de la disponibilité totale


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.5 - Utilisation des céréales</h3>
</div>

In [44]:
#Création d'une liste avec toutes les variables
# Liste manuelle des produits céréaliers
liste_cereales = [
    "Blé", "Riz (Eq Blanchi)", "Maïs", "Orge", "Seigle",
    "Avoine", "Sorgho", "Millet", "Céréales, Autres"
]

# Filtrage du DataFrame sur cette liste
cereales = dispo_pop[dispo_pop["Produit"].isin(liste_cereales)].copy()
pd.set_option('display.float_format', '{:,.0f}'.format)

display(cereales.head())


Zone           Produit   Origine  Aliments pour animaux  \
7   Afghanistan               Blé  vegetale                      0   
12  Afghanistan  Céréales, Autres  vegetale                      0   
32  Afghanistan              Maïs  vegetale                200,000   
34  Afghanistan            Millet  vegetale                      0   
40  Afghanistan              Orge  vegetale                360,000   

    Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
7                     0                                           1,369   
12                    0                                               0   
32                    0                                              21   
34                    0                                               3   
40                    0                                              26   

    Disponibilité alimentaire en quantité (kg/personne/an)  \
7                                                 160        
12                                                  0        
32                                                  2        
34                                                  0        
40                                                  3        

    Disponibilité de matière grasse en quantité (g/personne/jour)  \
7                                                   5               
12                                                  0               
32                                                  0               
34                                                  0               
40                                                  0               

    Disponibilité de protéines en quantité (g/personne/jour)  \
7                                                  37          
12                                                  0          
32                                                  1          
34                                                  0          
40                                                  1          

    Disponibilité intérieure  ...  Importations - Quantité  Nourriture  \
7                  5,992,000  ...                    1,173   4,895,000   
12                         0  ...                        0           0   
32                   313,000  ...                        1      76,000   
34                    13,000  ...                        0      12,000   
40                   524,000  ...                       10      89,000   

    Pertes  Production  Semences  Traitement  Variation de stock  Année  \
7  775,000       5,169   322,000           0                -350   2017   
12       0           0         0           0                   0   2017   
32  31,000         312     5,000           0                   0   2017   
34   1,000          13         0           0                   0   2017   
40  52,000         514    22,000           0                   0   2017   

    Population         dispo_kcal  
7   36,296,113 18,136,623,224,405  
12  36,296,113                  0  
32  36,296,113    278,209,706,145  
34  36,296,113     39,744,243,735  
40  36,296,113    344,450,112,370  

[5 rows x 21 columns]

In [45]:
#Création d'un dataframe avec les informations uniquement pour ces céréales
liste_cereales = [
    "Blé", "Riz (Eq Blanchi)", "Maïs", "Orge", "Seigle",
    "Avoine", "Sorgho", "Millet", "Céréales, Autres"
]

cereales = dispo_pop[dispo_pop["Produit"].isin(liste_cereales)].copy()

In [46]:
#Affichage de la proportion d'alimentation animale
display(cereales.head())

Zone           Produit   Origine  Aliments pour animaux  \
7   Afghanistan               Blé  vegetale                      0   
12  Afghanistan  Céréales, Autres  vegetale                      0   
32  Afghanistan              Maïs  vegetale                200,000   
34  Afghanistan            Millet  vegetale                      0   
40  Afghanistan              Orge  vegetale                360,000   

    Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
7                     0                                           1,369   
12                    0                                               0   
32                    0                                              21   
34                    0                                               3   
40                    0                                              26   

    Disponibilité alimentaire en quantité (kg/personne/an)  \
7                                                 160        
12                                                  0        
32                                                  2        
34                                                  0        
40                                                  3        

    Disponibilité de matière grasse en quantité (g/personne/jour)  \
7                                                   5               
12                                                  0               
32                                                  0               
34                                                  0               
40                                                  0               

    Disponibilité de protéines en quantité (g/personne/jour)  \
7                                                  37          
12                                                  0          
32                                                  1          
34                                                  0          
40                                                  1          

    Disponibilité intérieure  ...  Importations - Quantité  Nourriture  \
7                  5,992,000  ...                    1,173   4,895,000   
12                         0  ...                        0           0   
32                   313,000  ...                        1      76,000   
34                    13,000  ...                        0      12,000   
40                   524,000  ...                       10      89,000   

    Pertes  Production  Semences  Traitement  Variation de stock  Année  \
7  775,000       5,169   322,000           0                -350   2017   
12       0           0         0           0                   0   2017   
32  31,000         312     5,000           0                   0   2017   
34   1,000          13         0           0                   0   2017   
40  52,000         514    22,000           0                   0   2017   

    Population         dispo_kcal  
7   36,296,113 18,136,623,224,405  
12  36,296,113                  0  
32  36,296,113    278,209,706,145  
34  36,296,113     39,744,243,735  
40  36,296,113    344,450,112,370  

[5 rows x 21 columns]

In [47]:
#Affichage de la proportion d'alimentation animale

# Somme de la disponibilité intérieure totale pour les céréales
total_cereales = cereales["Disponibilité intérieure"].sum()

# Somme de la quantité utilisée pour les animaux
alim_animale = cereales["Aliments pour animaux"].sum()

# Calcul du pourcentage
proportion_animaux = (alim_animale / total_cereales) * 100

print(f"Proportion des céréales utilisées pour l'alimentation animale : {proportion_animaux:.2f}%")

Proportion des céréales utilisées pour l'alimentation animale : 36.14%


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.6 - Pays avec la proportion de personnes sous-alimentée la plus forte en 2017</h3>
</div>

In [ ]:
# On part de popu_nutri, et on garde 2017 si présent
p2017 = popu_nutri.copy()
if "Année" in p2017.columns:
    p2017 = p2017[p2017["Année"] == 2017]

# Valeurs numériques + filtrage de base
p2017[["Population","sous_nutrition"]] = p2017[["Population","sous_nutrition"]].apply(pd.to_numeric, errors="coerce")
p2017 = p2017.query("Population > 0 and sous_nutrition > 0").copy()

# Calcul le pourcentage
p2017 = p2017[p2017["sous_nutrition"] != 100]
p2017["Proportion_sous_nutrition"] = (p2017["sous_nutrition"] * 1000 / p2017["Population"]) * 100

In [49]:
#affichage après trie des 10 pires pays
cols = ["Zone", "Population", "sous_nutrition", "Proportion_sous_nutrition"]
pays_pires = (
    p2017.sort_values("Proportion_sous_nutrition", ascending=False)
         .loc[:, cols].head(10).reset_index(drop=True)
)
pays_pires["Proportion_sous_nutrition"] = pays_pires["Proportion_sous_nutrition"].round(2).map("{:.2f}%".format)
display(pays_pires)

Zone  Population  sous_nutrition  \
0                                       Haïti  10,982,366           5,300   
1  République populaire démocratique de Corée  25,429,825          12,000   
2                                  Madagascar  25,570,512          10,500   
3                                     Libéria   4,702,226           1,800   
4                                     Lesotho   2,091,534             800   
5                                       Tchad  15,016,753           5,700   
6                                      Rwanda  11,980,961           4,200   
7                                  Mozambique  28,649,018           9,400   
8                                 Timor-Leste   1,243,258             400   
9                                 Afghanistan  36,296,113          10,500   

  Proportion_sous_nutrition  
0                    48.26%  
1                    47.19%  
2                    41.06%  
3                    38.28%  
4                    38.25%  
5                    37.96%  
6                    35.06%  
7                    32.81%  
8                    32.17%  
9                    28.93%

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.7 - Pays qui ont le plus bénéficié d'aide alimentaire depuis 2013</h3>
</div>

In [50]:
#calcul du total de l'aide alimentaire par pays

# On regroupe par pays et on additionne la quantité d'aide reçue (en kg)
aide_par_pays = aide.groupby("Zone")["Valeur"].sum().reset_index()

# On renomme la colonne pour plus de clarté
aide_par_pays = aide_par_pays.rename(columns={"Valeur": "Aide_totale"})

# Vérification rapide
display(aide_par_pays.head())

Zone  Aide_totale
0  Afghanistan    185452000
1      Algérie     81114000
2       Angola      5014000
3   Bangladesh    348188000
4      Bhoutan      2666000

In [51]:
#Affichage après trie des 10 pays qui ont bénéficié le plus de l'aide alimentaire

# Tri décroissant selon la quantité d’aide reçue
top_aide = aide_par_pays.sort_values(by="Aide_totale", ascending=False)

# Affichage des 10 premiers pays
display(top_aide.head(10))

Zone  Aide_totale
50         République arabe syrienne   1858943000
75                          Éthiopie   1381294000
70                             Yémen   1206484000
61                     Soudan du Sud    695248000
60                            Soudan    669784000
30                             Kenya    552836000
3                         Bangladesh    348188000
59                           Somalie    292678000
53  République démocratique du Congo    288502000
43                             Niger    276344000

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.8 - Evolution des 5 pays qui ont le plus bénéficiés de l'aide alimentaire entre 2013 et 2016</h3>
</div>

In [52]:
#Création d'un dataframe avec la zone, l'année et l'aide alimentaire puis groupby sur zone et année
aide_2013_2016 = aide[aide["Année"].between(2013, 2016)].copy()

# 1) agrégat par Zone & Année
cinq_pays_aide = (
    aide_2013_2016[["Zone", "Année", "Valeur"]]
    .groupby(["Zone", "Année"], as_index=False)
    .sum()
    .rename(columns={"Valeur": "Aide_Alimentaire"})
)

In [ ]:
#Création d'une liste contenant les 5 pays qui ont le plus bénéficié de l'aide alimentaire
liste_pays = ["République arabe syrienne", "Éthiopie", "Yémen", "Soudan du Sud", "Soudan"]

In [54]:
#On filtre sur le dataframe avec notre liste
cinq_pays_aide = cinq_pays_aide.loc[cinq_pays_aide["Zone"].isin(liste_pays), :]

In [64]:
#Affichage des pays avec l'aide alimentaire par année
cinq_pays_aide

Zone  Année  Aide_Alimentaire
157  République arabe syrienne   2013         563566000
158  République arabe syrienne   2014         651870000
159  République arabe syrienne   2015         524949000
160  République arabe syrienne   2016         118558000
189                     Soudan   2013         330230000
190                     Soudan   2014         321904000
191                     Soudan   2015          17650000
192              Soudan du Sud   2013         196330000
193              Soudan du Sud   2014         450610000
194              Soudan du Sud   2015          48308000
214                      Yémen   2013         264764000
215                      Yémen   2014         103840000
216                      Yémen   2015         372306000
217                      Yémen   2016         465574000
225                   Éthiopie   2013         591404000
226                   Éthiopie   2014         586624000
227                   Éthiopie   2015         203266000

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.9 - Pays avec le moins de disponibilité par habitant</h3>
</div>

In [56]:
# Calcul de la disponibilité en kcal par personne par jour par pays
dispo_kcal = (
    dispo_pop[["Zone", "Disponibilité alimentaire (Kcal/personne/jour)"]]
      .groupby("Zone", as_index=False)
      .sum(min_count=1)
      .rename(columns={"Disponibilité alimentaire (Kcal/personne/jour)": "Kcal/personne/jour"})
)

In [57]:
# Affichage des 10 pays qui ont le moins de dispo alimentaire par personne
pays_moins_dispo = (
    dispo_kcal
      .sort_values("Kcal/personne/jour", ascending=True)
      .head(10)
      .reset_index(drop=True)
)
display(pays_moins_dispo)

Zone  Kcal/personne/jour
0                   République centrafricaine               1,879
1                                      Zambie               1,924
2                                  Madagascar               2,056
3                                 Afghanistan               2,087
4                                       Haïti               2,089
5  République populaire démocratique de Corée               2,093
6                                       Tchad               2,109
7                                    Zimbabwe               2,113
8                                     Ouganda               2,126
9                                 Timor-Leste               2,129

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.10 - Pays avec le plus de disponibilité par habitant</h3>
</div>

In [58]:
# Calcul de la disponibilité totale en kg/personne/an par pays
pays_plus_dispo = (
    dispo_kcal
      .sort_values("Kcal/personne/jour", ascending=False)
      .head(10)
      .reset_index(drop=True)
)
display(pays_plus_dispo)

Zone  Kcal/personne/jour
0               Autriche               3,770
1               Belgique               3,737
2                Turquie               3,708
3  États-Unis d'Amérique               3,682
4                 Israël               3,610
5                Irlande               3,602
6                 Italie               3,578
7             Luxembourg               3,540
8                 Égypte               3,518
9              Allemagne               3,503

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.11 - Exemple de la Thaïlande pour le Manioc</h3>
</div>

In [59]:
#création d'un dataframe avec uniquement la Thaïlande 
thai_manioc = (
    dispo_pop.query("Zone == 'Thaïlande'")
             .loc[lambda d: d["Produit"].str.contains("Manioc", case=False, na=False)]
             .copy()
)

display(thai_manioc[[
    "Zone", "Produit", "Production",
    "Exportations - Quantité",
    "Importations - Quantité",
    "Disponibilité intérieure"
]])

Zone Produit  Production  Exportations - Quantité  \
13620  Thaïlande  Manioc      30,228                   25,214   

       Importations - Quantité  Disponibilité intérieure  
13620                    1,250                 6,264,000

In [60]:
# Calcul de la sous-nutrition en Thaïlande
sn_thailande = sous_nutrition[sous_nutrition["Zone"] == "Thaïlande"]

# Si des données existent
if not sn_thailande.empty:
    # Valeur en millions (valeurs d'origine en milliers)
    valeur_millions = sn_thailande["sous_nutrition"].iloc[-1] / 1000

    # Pourcentage estimé sur base d'environ 70 millions d'habitants
    pourcentage = (valeur_millions / 70) * 100

    print(f"Sous-nutrition en Thaïlande : {valeur_millions:.2f} millions de personnes")
    print(f"≈ {pourcentage:.2f}% de la population")
else:
    print("⚠️ Pas de donnée de sous-nutrition pour la Thaïlande.")

Sous-nutrition en Thaïlande : 6.50 millions de personnes
≈ 9.29% de la population


In [61]:
# On calcule la proportion exportée en fonction de la proportion
thai_manioc["Proportion exportée (%)"] = (
    thai_manioc["Exportations - Quantité"] / thai_manioc["Production"] * 100
).round(2)

display(thai_manioc[[
    "Zone", "Produit", "Production",
    "Exportations - Quantité", "Proportion exportée (%)"
]])

Zone Produit  Production  Exportations - Quantité  \
13620  Thaïlande  Manioc      30,228                   25,214   

       Proportion exportée (%)  
13620                       83

<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Etape 6 - Analyse complémentaires</h2>
</div>

In [62]:
#ajouter en dessous toutes les analyses complémtaires